qiskit_nature/results/protein_folding_result.py

```diff
@@ -9,54 +9,146 @@
 # Any modifications or derivative works of this code must retain this
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
+
 """The protein folding result."""
-from typing import Union
 
-from qiskit.opflow import PauliOp
+from typing import List, Optional
+from qiskit.utils import optionals as _optionals
+from ..problems.sampling.protein_folding.peptide.peptide import Peptide
+from .eigenstate_result import EigenstateResult
+from .utils.protein_plotter import ProteinPlotter
+from .utils.protein_shape_decoder import ProteinShapeDecoder
+from .utils.protein_shape_file_gen import ProteinShapeFileGen
+
 
-from qiskit_nature.problems.sampling.protein_folding.protein_folding_problem import (
-    ProteinFoldingProblem,
-)
-from qiskit_nature.results import EigenstateResult
+if _optionals.HAS_MATPLOTLIB:
+    # pylint: disable=import-error,unused-import
+    from matplotlib.pyplot import figure
 
 
 class ProteinFoldingResult(EigenstateResult):
-    """The protein folding result."""
+    """
+    The Protein Folding Result.
+    This class interprets a bitstring encoding the turns of a protein from
+    :class:`~qiskit_nature.problems.sampling.protein_folding_problem.ProteinFoldingProblem`
+    and decodes it. One can generate a .xyz file
+    (using :meth:`~qiskit_nature.results.ProteinFoldingResult.save_xyz_file`), which is a file
+    containing the cartesian coordinates of each atom in the protein. This kind of file can be
+    used with other software to generate plots of the molecule.
+    Alternatively, one can use :meth:`~qiskit_nature.results.ProteinFoldingResult.get_figure`.
+    Note that `matplotlib` needs to be installed in order to generate such a figure.
+    """
 
     def __init__(
         self,
-        protein_folding_problem: ProteinFoldingProblem,
-        best_sequence: Union[str, PauliOp],
+        peptide: Peptide,
+        unused_qubits: List[int],
+        turns_sequence: str,
```

Comment:
```suggestion
        turn_sequence: str,
```
Personally, I find this name a bit better